internal/app/wwctl/overlay/imprt/main.go

```diff
@@ -9,39 +9,44 @@ import (
 	"github.com/hpcng/warewulf/internal/pkg/overlay"
 	"github.com/hpcng/warewulf/internal/pkg/util"
 	"github.com/hpcng/warewulf/internal/pkg/wwlog"
+	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 )
 
 func CobraRunE(cmd *cobra.Command, args []string) error {
-	overlayName := args[0]
-	source := args[1]
 	var dest string
 	var overlaySource string
 
-	if len(args) == 3 {
-		dest = args[2]
+	overlayKind := args[0]
+	overlayName := args[1]
+	source := args[2]
+
+	if overlayKind != "system" && overlayKind != "runtime" {
+		return errors.New("overlay kind must be of type 'system' or 'runtime'")
+	}
+
+	if len(args) == 4 {
+		dest = args[3]
 	} else {
 		dest = source
 	}
 
-	if SystemOverlay {
-		wwlog.Printf(wwlog.VERBOSE, "Importing '%s' into system overlay '%s:%s'\n", source, overlayName, dest)
+	if overlayKind == "system" {
+		wwlog.Printf(wwlog.VERBOSE, "Copying '%s' into system overlay '%s:%s'\n", source, overlayName, dest)
 		overlaySource = config.SystemOverlaySource(overlayName)
-	} else {
-		wwlog.Printf(wwlog.VERBOSE, "Importing '%s' into runtime overlay '%s:%s'\n", source, overlayName, dest)
+	} else if overlayKind == "runtime" {
+		wwlog.Printf(wwlog.VERBOSE, "Copying '%s' into runtime overlay '%s:%s'\n", source, overlayName, dest)
 		overlaySource = config.RuntimeOverlaySource(overlayName)
 	}
 
 	if !util.IsDir(overlaySource) {
-		wwlog.Printf(wwlog.ERROR, "Overlay does not exist: %s\n", overlayName)
+		wwlog.Printf(wwlog.ERROR, "Overlay does not exist: %s:%s\n", overlayKind, overlayName)
 		os.Exit(1)
 	}
 
 	err := util.CopyFile(source, path.Join(overlaySource, dest))
 	if err != nil {
-		wwlog.Printf(wwlog.ERROR, "Failed copying file into overlay sourcedir:\n")
-		wwlog.Printf(wwlog.ERROR, "%s\n", err)
-		os.Exit(1)
+		return errors.Wrap(err, "could not copy file into overlay")
```

Comment:
For user friendliness, if a user only provides a directory as a destination, could we take the base filename of the source and use that as the target filename?

Just tested this out:
```
if util.IsDir(path.Join(overlaySource, dest)) {
    dest = path.Join(dest, path.Base(source))
}
```
One other thing, if there is a file in the dest with the same name, should we warn the user about it and exit?
Just tested this below the code I suggested above.
```
 if util.IsFile(path.Join(overlaySource, dest)) {
    wwlog.Printf(wwlog.ERROR, "A file with that name already exists in the %s overlay %s\n:", overlayKind, overlayName)
    os.Exit(1)
}
```

Reply:
Import should indeed take the source path as the destination as this is "supposed" to do:
```
	if len(args) == 4 {
		dest = args[3]
	} else {
		dest = source
	}
```

Is your first suggestion a replacement to this or another conditional?

Regarding the second bit of code, totally agree. Will throw that in there and commit what I have.

Reply:
That block takes care of when the user does not specify a destination at all.

Example:
`sudo wwctl overlay import runtime myOverlay1 /home/myoung/warewulf-4.0.0-1.el8.x86_64.rpm`

The resulting path passed to util.CopyFile() is, "/var/warewulf/overlays/runtime/myOverlay1/home/myoung/warewulf-4.0.0-1.el8.x86_64.rpm".

I am suggesting adding another conditional to handle when a user supplies only a directory name as the destination, without specifying the file name.

Example
`sudo wwctl overlay import runtime myOverlay1 /home/myoung/warewulf-4.0.0-1.el8.x86_64.rpm /testing09142021/`

Right now, util.CopyFile() errors out saying that the file could not be copied because 'testing09142021' is a directory.

The path.Join() on line 47, `err := util.CopyFile(source, path.Join(overlaySource, dest))`,  results in the path being "/var/warewulf/overlays/runtime/myOverlay1/testing09142021" which is indeed a directory.

My initial flow when I was testing was to use mkdir first and then import a file into that directory without having to type out or copy and paste the file name. If a user were to use the alias, cp, instead of import, it feels like a natural thing to do.

Reply:
Ahhhhh, very good call, and thank you for the explanation!